### PR TITLE
Lesson 7

### PR DESCRIPTION
**Pull request by Martochkaa**

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -116,4 +112,648 @@ let chahgeColor3 = f1();
...
+        a: 'a',
+      },
+      b: 'b',
+      a: 'a',
+    },
+    b: 'b',
+ };
+
+ function deepEqual(obj1, obj2) { // Цикл через свойства объекта obj1
+    for (var p in obj1) { //Проверка на то, что оба объекта существуют
```
Comment (Dronom) [inline main.js]: don't use var

File: main.js [hunk 2]
```diff
@@ -116,4 +112,648 @@ let chahgeColor3 = f1();
...
+          // Сравнение данных типа function:
+        case 'function':
+          if (typeof (obj2[p]) == 'undefined' || (p != 'compare' && obj1[p].toString() != obj2[p].toString())) return false;
+          break;  // Сравнение значений:
+        default:
+          if (obj1[p] != obj2[p]) return false;
+      }
+    }
+    return true;
+  }
```
Comment (Dronom) [inline main.js]: try call deepEqual(obj2,obj3)

Comment (Martochkaa) [reply main.js:756]: Does not work. Do not understand why. Can you hear an explanation?

**Review by Dronom — APPROVED** [review]
Looks good

**Pull request merged**